src/vs/workbench/api/node/mainThreadEditors.ts

```diff
@@ -102,11 +104,16 @@ export class MainThreadEditors extends MainThreadEditorsShape {
 
 	// --- from extension host process
 
-	$tryShowTextDocument(resource: URI, position: EditorPosition, preserveFocus: boolean): TPromise<string> {
+	$tryShowTextDocument(resource: URI, options: vscode.ShowTextDocumentOptions): TPromise<string> {
+		const position: EditorPosition = TypeConverters.fromViewColumn(options.column);
```

Comment:
The main issue with that, is then we would need 2 "versions" of the `TextDocument[Show|View|Display]Options` -- one for the extensions with a `column` and one for the main-side with a `position`. Which is certainly doable, I was just trying to not have 2 types with almost the same fields. Are you fine with adding 2 types? And if so, where should the main-side one live and how should it be named to not cause confusion?